### PR TITLE
Chore: Convert app to use shakapacker instead of webpacker

### PR DESCRIPTION
**Pull request by CouchofTomato**
Because:
Webpacker is no longer being maintained, and our app still relies on
the sassc gem, which prevents upgrading to jsbundling and cssbundling
gems, we need to convert our app to use shakapacker.

This commit:

* Update webpacker source entry path to top level javascript folder
* Move top level javascript files from packs to javascript directory
* Change webpacker gem to shakapacker 6.4.1
* Add shakapacker 6.4.1 npm package
* Remove env specific webpack config files
* Delete browserlistrc file
* Add heroku release binstub
* Add webpacker and webpacker-dev-server binstubs
* Add yarn binstub
* Update content security policy
* Add webpack additional configuration for css tooling
* Update webpacker yml file to work with shakapacker defaults
* Install required npm packages for shakapacker

Complete the following REQUIRED checkboxes:

-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [ ] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behaviour
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes

Complete the following checkbox ONLY IF it is applicable to your PR. You can complete it later if it is not currently applicable:
-   [ ] I have verified all tests and linters pass against my changes, and/or I have included automated tests where applicable

<hr>

**1. Because:**

Closes #XXXXX

**2. This PR:**

**3. Additional Information:**

### CODE REVIEW

**Review by leosoaivan — APPROVED** [review]

File: config/webpacker.yml [hunk 1]
```diff
@@ -49,32 +47,37 @@ default: &default
...
+    # live_reload: true
+    client:
+      # Should we show a full-screen overlay in the browser when there are compiler errors or warnings?
+      overlay: true
+      # May also be a string
+      # webSocketURL:
+      #  hostname: "0.0.0.0"
+      #  pathname: "/ws"
+      #  port: 8080
+    # Should we use gzip compression?
```
Comment (KevinMulhern) [inline config/webpacker.yml:71]: Were these comments added by the webpacker install?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
+    <% content_for :error_tracking do %>
```
Comment (KevinMulhern) [inline app/views/layouts/application.html.erb:4]: Why do we need to use a content tag here? does it have something to do with the `append_javascript_pack_tag` helper?

File: config/webpacker.yml (hunk 1 above)
Comment (CouchofTomato) [reply config/webpacker.yml:71]: Yeah. That's all default shakapacker stuff mate.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (CouchofTomato) [reply app/views/layouts/application.html.erb:4]: You'll see it's the recommended approach here https://github.com/shakacode/shakapacker#view-helper-append_javascript_pack_tag-and-append_stylesheet_pack_tag

The alternative is we can move error tracking up into the head and render it there but `append_javascript_pack_tag` must come before `javascript_pack_tag`

Comment (KevinMulhern) [reply app/views/layouts/application.html.erb:4]: Gotcha, thanks for those links mate.

**Review by KevinMulhern — APPROVED** [review]
Looks great to me @CouchofTomato, it's snappier in development and should hopefully speed up the asset compile phase of deploys 🚀 

thanks for the work you've put into this 💪

**Pull request merged**